plugins/modules/load_balancer.py

```
@@ -0,0 +1,365 @@
+        else:
+            load_balancer['state'] = 'absent'
+        return load_balancer
+
+    # Create LB
+    def _create_load_balancer(self, load_balancer_info):
+        self._result['changed'] = True
+
+        data = deepcopy(self._module.params)
+        for i in ('state', 'api_timeout', 'api_token', 'api_url'):
+            del data[i]
+
+        self._result['diff']['before'] = self._init_load_balancer_container()
+        self._result['diff']['after'] = deepcopy(data)
+        if not self._module.check_mode:
+            self._post('load-balancers', data)
+            load_balancer_info = self._wait_for_state(('running', ))
+        return load_balancer_info
+
+    # Wait for LB to be running
+    def _wait_for_state(self, states):
+        start = datetime.now()
+        timeout = self._module.params['api_timeout'] * 2
+        while datetime.now() - start < timedelta(seconds=timeout):
+            load_balancer_info = self._get_load_balancer_info(refresh=True)
+            if load_balancer_info.get('state') in states:
+                return load_balancer_info
+            sleep(1)
+
+        # Timeout succeeded
+        if load_balancer_info.get('name') is not None:
+            msg = "Timeout while waiting for a state change on load balancer %s to states %s. " \
+                  "Current state is %s." % (load_balancer_info.get('name'), states, load_balancer_info.get('state'))
+        else:
+            name_uuid = self._module.params.get('name') or self._module.params.get('uuid')
+            msg = 'Timeout while waiting to find the load balancer %s' % name_uuid
+
+        self._module.fail_json(msg=msg)
+
+    # Update LB
+    def _update_load_balancer(self, load_balancer_info):
+
+        previous_state = load_balancer_info.get('state')
+
+        load_balancer_info = self._update_param('name', load_balancer_info)
+        # load_balancer_info = self._update_param('flavor', load_balancer_info, requires_stop=True)     # Not yet added to API
```

This won't be added in the near future. 

How do we handle immutable properties in other modules? I would handle this the same way. Probably either silently ignore it, issue a warning or abort.  A mismatch of the vip_addresses should be handled the same way.